practice/pig-latin.py

```diff
@@ -1,58 +1,122 @@
 Note: With these goals, I'd like to avoid ruining punctuation around short words such as I and My.
+EDIT 12-July-2016: Altering code to be a series of functions, solved for some apostrophes and capitals.
+EDIT 12-July-2016: Added list of words in English that start with Y-vowel for higher accuracy.
 """
 
-# 1. Setup
-FIRST_VOWEL = 'a e i o u'.split()
-LATER_VOWEL = 'a e i o u y'.split()
+
+# 1. Setup Define/Functions
+
+
+FIRST_VOWEL = 'a e i o u A E I O U'.split()
+LATER_VOWEL = 'a e i o u y A E I O U Y'.split()
+SPECIAL_CASE = 'Ybe ybe Ycleped ycleped Yclept yclept Ysgard Ydo ydo Ydrad ydrad Yfere yfere Ygdrasyl ygdrasyl Yghe'\
+               'ygh Ygo ygo Yground yground Ypres ypres Ypsiliform ypsiliform Ypsiloid ypsiloid Yren yren Ysame ysame '\
+               'Ytterbic ytterbic Ytterbium ytterbium Yttria yttria Yttric yttric Yttriferous yttriferous Yttrious ' \
+               'yttrious Yttrium yttrium Yttro-cerite yttro-cerite Yttro-columbite yttro-columbite Yttro-tantalite ' \
+               'yttro-tantalite Yvel yvel Ywar ywar Ywis ywis'.split()
 PUNCTUATION_LIST = list(string.punctuation)
 CAPITALS_LIST = list(string.ascii_uppercase)
-new_punctuation = ''
-new_phrase = ''
 
-# 2. Input
 
-english_phrase = input("What would you like to Anslatetray? ").split()
+# 2. Define Functions
+
+
+def get_phrase_casted_into_list():
+    """Aquires english phrase from user input and casts it as a variable"""
+    typed_phrase = input("What would you like to Anslatetray? ").split()
+    return typed_phrase
+
+
+def pull_english_word_from_list (phrase, i):
+    """Pulls a single word from the list of english words to be translated."""
+    pull_word = phrase[i]
+    return pull_word
+
+def flip_capital_boolean(word):
+    """Checks for capitals at the start of a word, flips a boolean switch to run capitalizing function"""
+    if word[0] in CAPITALS_LIST:
+        return True
+    else:
+        return False
 
-# 3. Transform
+def check_for_and_cut_capitals(english_word):
+    """Checks for capitals at the start of a word, and casts it to lowercase, checking capitalize_boolean true"""
+    uppercase_first_letter = english_word[0]
+    lowercase_first_lettter = uppercase_first_letter.lower()
+    strip_capital_english_word = english_word.replace(uppercase_first_letter,lowercase_first_lettter,1)
+    return strip_capital_english_word
 
-i= 0
-while i < len(english_phrase):
-    english_word = english_phrase[i]
-    if english_word[-1] in PUNCTUATION_LIST:
-        new_punctuation = english_word[-1]
-        modified_english_word = english_word.strip(new_punctuation)
-        re_capitlaize_word = False
-    elif english_word[0] in CAPITALS_LIST:
-        modified_english_word = english_word.lower()
-        re_capitlaize_word = True
+def flip_punctuation_boolean(word):
+    """Checks for punctuation at end of word and flips a boolean switch if it is present """
+    if word[-1] in PUNCTUATION_LIST:
```

Comment:
When you have something that looks like:

```
if ...:
    return True
else:
    return False
```

You could more simply write `return ...` so in this case `return not word[-1] in PUNCTUATION_LIST`.